core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigBuildSteps.java

```diff
@@ -0,0 +1,95 @@
+package io.quarkus.deployment.steps;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+import org.eclipse.microprofile.config.spi.Converter;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.DeploymentClassLoaderBuildItem;
+import io.quarkus.deployment.builditem.GeneratedClassBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationSourceBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ServiceProviderBuildItem;
+import io.quarkus.deployment.util.ServiceUtil;
+import io.quarkus.gizmo.ClassCreator;
+import io.quarkus.gizmo.ClassOutput;
+import io.quarkus.gizmo.MethodCreator;
+import io.quarkus.gizmo.MethodDescriptor;
+import io.quarkus.gizmo.ResultHandle;
+import io.smallrye.config.SmallRyeConfigProviderResolver;
+
+class ConfigBuildSteps {
+
+    static final String PROVIDER_CLASS_NAME = "io.quarkus.runtime.generated.ConfigSourceProviderImpl";
+
+    static final String SERVICES_PREFIX = "META-INF/services/";
+
+    @BuildStep
+    void generateConfigSources(List<RunTimeConfigurationSourceBuildItem> runTimeSources,
+            final BuildProducer<GeneratedClassBuildItem> generatedClass) {
+        ClassOutput classOutput = new ClassOutput() {
```

Comment:
I think you can just use io.quarkus.deployment.GizmoAdaptor. Looks like there are a few places in the code base that have this exact same anonymous class.

Reply:
Ah, I just copied it from... somewhere... once the other `ClassOutput` was gone I thought the adapter was too.   I can update that.

Reply:
Fixed.